units/pod_definition_creation.go

```diff
@@ -0,0 +1,206 @@
+	if err != nil {
+		j.AddRetryableError(errors.Wrapf(err, "finding dependent intent pods with family '%s'", j.Family))
+		return
+	}
+	if len(dependents) == 0 {
+		// No-op if there are no pods that need this definition to be created.
+		return
+	}
+
+	podDefOpts, err := cloud.ExportECSPodDefinitionOptions(&j.settings, j.ContainerOpts)
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "creating pod definition with family '%s'", j.Family))
+		return
+	}
+
+	podDef, err := definition.FindOneByFamily(j.Family)
+	if err != nil {
+		j.AddRetryableError(errors.Wrapf(err, "checking for existing pod definition with family '%s'", j.Family))
+		return
+	}
+	if podDef != nil {
+		// No-op if this pod definition already exists.
+		return
+	}
+
+	item, err := j.ecsPodDefManager.CreatePodDefinition(ctx, *podDefOpts)
+	if err != nil {
+		j.AddRetryableError(errors.Wrapf(err, "creating pod definition with family '%s'", j.Family))
+		return
+	}
+
+	grip.Info(message.Fields{
+		"message":     "successfully created pod definition",
+		"family":      j.Family,
+		"external_id": item.ID,
+		"job":         j.ID(),
+	})
+}
+
+func (j *podDefinitionCreationJob) populateIfUnset(ctx context.Context) error {
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+
+	// Use the latest service flags instead of those cached in the environment.
+	settings := *j.env.Settings()
+	if err := settings.ServiceFlags.Get(j.env); err != nil {
+		return errors.Wrap(err, "getting service flags")
+	}
+	j.settings = settings
+
+	if j.vault == nil {
+		if j.smClient == nil {
+			client, err := cloud.MakeSecretsManagerClient(&settings)
+			if err != nil {
+				return errors.Wrap(err, "initializing Secrets Manager client")
+			}
+			j.smClient = client
+		}
+		vault, err := cloud.MakeSecretsManagerVault(j.smClient)
+		if err != nil {
+			return errors.Wrap(err, "initializing Secrets Manager vault")
+		}
+		j.vault = vault
+	}
+
+	if j.ecsClient == nil {
+		client, err := cloud.MakeECSClient(&settings)
+		if err != nil {
+			return errors.Wrap(err, "initializing ECS client")
+		}
+		j.ecsClient = client
+	}
+
+	if j.ecsPodDefManager == nil {
+		podDefMgr, err := cloud.MakeECSPodDefinitionManager(j.ecsClient, j.vault)
+		if err != nil {
+			return errors.Wrap(err, "initializing ECS pod creator")
+		}
+		j.ecsPodDefManager = podDefMgr
+	}
+
+	return nil
+}
+
+// decommissionDependentIntentPods decommissions all intent pods that depend on
+// the pod definition created by this job.
+func (j *podDefinitionCreationJob) decommissionDependentIntentPods() error {
+	podsToDecommission, err := pod.FindIntentByFamily(j.Family)
+	if err != nil {
+		return errors.Wrap(err, "finding intent pods to decommission")
+	}
+	catcher := grip.NewBasicCatcher()
+	var podIDs []string
+	for _, p := range podsToDecommission {
+		catcher.Wrapf(p.UpdateStatus(pod.StatusDecommissioned), "pod '%s'", p.ID)
+		podIDs = append(podIDs, p.ID)
+	}
+
+	grip.Info(message.Fields{
```

Comment:
Do we want to skip this logging if podIDs is empty?

Reply:
Done.